routes/api/contacts.js

```diff
@@ -1,25 +1,155 @@
 
-const router = express.Router()
+const { listContacts, getContactById, addContact, removeContact, updateContact } = require("../../models/contacts.js");
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const contactSchema = Joi.object({
+	name: Joi.string().required(),
+	email: Joi.string().email().required(),
+	phone: Joi.string().required(),
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+	try {
+		const contacts = await listContacts();
+		res.json({
+			status: "success",
+			code: 200,
+			results: contacts.length,
+			data: {
+				contacts,
+			},
+		});
+	} catch (error) {
+		console.error("Error reading file: ", error.message);
+		next(error);
+	}
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+	try {
+		const contactId = await req.params.contactId;
+		const contact = await getContactById(contactId);
+		if (contact) {
+			res.json({
+				status: "success",
+				code: 200,
+				data: {
+					contact,
+				},
+			});
+		} else {
+			res.status(404).json({
+				status: "fail",
+				message: "Not found",
+			});
+		}
+	} catch (error) {
+		next(error);
+	}
+});
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+	try {
+		const { name, email, phone } = await req.body;
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+		if (!name || !email || !phone) {
+			return res.status(400).json({
+				status: "fail",
+				message: "Missing required fields: name, email, phone",
+			});
+		}
 
-module.exports = router
+		const validateNewContact = contactSchema.validate({ name, email, phone });
+
+		if (validateNewContact.error) {
+			return res.status(400).json({
+				status: "fail",
+				message: "Invalid data",
+				error: validateNewContact.error,
+			});
+		}
+
+		const newContact = await addContact({ name, email, phone });
+
+		res.json({
+			status: "success",
+			code: 201,
+			data: {
+				newContact,
+			},
+		});
+	} catch (error) {
+		next(error);
+	}
+});
+
+router.delete("/:contactId", async (req, res, next) => {
+	try {
+		const contactId = await req.params.contactId;
+		const contact = await removeContact(contactId);
+		if (contact) {
+			res.json({
+				status: "success",
+				code: 200,
+				message: "contact deleted",
+				data: {
+					contact,
+				},
+			});
+		} else {
+			res.status(404).json({
+				status: "fail",
+				message: "Not found",
+			});
+		}
+	} catch (error) {
+		next(error);
+	}
+});
+
+router.put("/:contactId", async (req, res, next) => {
+	try {
+		const contactId = req.params.contactId;
+		const { name, email, phone } = await req.body;
+
+		if (!name || !email || !phone) {
```

Comment:
To samo co z post + dodał bym odzielną scheme do walidacji put i tu np bez required() bo możemy np chcieć updateować tylko jeden atrybut. Pamiętaj też wtedy w updateContact przekazać cały obiekt
```
const validateNewContact = updateContactSchema.validate(req.body);
// if error ....
const updateData = await updateContact(contactId, req.body);
```

Reply:
Poprawione